internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -101,78 +103,108 @@ func (r *ClusterDeploymentsReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
+	clusterInstall := &hiveext.AgentClusterInstall{}
+	if clusterDeployment.Spec.ClusterInstallRef == nil {
+		r.Log.Infof("AgentClusterInstall not set for ClusterDeployment %s", clusterDeployment.Name)
```

Comment:
@avishayt, should this case be somehow reflected in conditions? for usability

Reply:
I guess "synced" should be set to false?

Reply:
I don't have an object to set conditions on.
We are not updating ClusterDeployment anymore

Reply:
Conditions would now be set on the AgentClusterInstall. We require those handful documented in the enhancement, but you can add your own.